test/integration/controller/jobset_controller_test.go

```diff
@@ -221,18 +236,21 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 			// Perform a series of updates to jobset resources and check resulting jobset state after each update.
 			for _, update := range tc.updates {
 
-				// Fetch updated job objects so we always have the latest resource versions to perform mutations on.
-				var jobList batchv1.JobList
-				gomega.Expect(k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace))).Should(gomega.Succeed())
+				var jobSet jobset.JobSet
+				gomega.Eventually(k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &jobSet), timeout, interval).Should(gomega.Succeed())
 
-				// Perform mutation on jobset if specified.
-				if update.jobUpdateFn != nil {
+				if update.jobSetUpdateFn != nil {
+					update.jobSetUpdateFn(&jobSet)
+				} else if update.jobUpdateFn != nil {
```

Comment:
Why is this `else if` instead of just `if`? Wouldn't it be more flexible to allow test cases to perform both a jobset update and then update job(s) in the same iteration? Or is there a reason these must be mutually exclusive?

Reply:
@ahg-g suggested that we should only really do either job or jobset updates.

Reply:
https://github.com/kubernetes-sigs/jobset/pull/20/files#r1176494351